src/BlockTypes/ProductGallery.php

```diff
@@ -14,6 +14,45 @@ class ProductGallery extends AbstractBlock {
 	 */
 	protected $block_name = 'product-gallery';
 
+	/**
+	 * Return the dialog content.
+	 *
+	 * @return string
+	 */
+	protected function render_dialog() {
+		$template_part = BlockTemplateUtils::get_template_part( 'product-gallery' );
+
+		$parsed_template = parse_blocks(
+			$template_part
+		);
+
+		$html = array_reduce(
+			$parsed_template,
+			function( $carry, $item ) {
+				return $carry . render_block( $item );
+			},
+			''
+		);
+
+		$gallery_dialog = '<dialog data-wc-bind--open="selectors.woocommerce.isDialogOpen">' . $html . '</dialog>';
+		return $gallery_dialog;
+	}
+
+
+	/**
+	 * This function remove the div wrapper.
+	 * The content has a <div> with the class wp-block-woocommerce-product-gallery>.
+	 * We don't need since that we add it in the render method.
+	 *
+	 * @param string $content Block content.
+	 * @return string Rendered block type output.
+	 */
+	private function remove_div_wrapper( $content ) {
+		$parsed_string = preg_replace( '/<div class="wp-block-woocommerce-product-gallery">/', '', $content );
+		$parsed_string = preg_replace( '/<\/div>$/', '', $parsed_string );
+		return $parsed_string;
+	}
```

Comment:
This seems a but fragile.

Can we perhaps utilize `DOMDocument` to parse this in a more robust fashion?

Reply:
I agree. Unfortunately, DOMDocument on performance level isn't the best since it has to parse the entire HTML code 😞. I don't see any better solution.